engine-standalone-tracing/src/types/mod.rs

```diff
@@ -319,7 +319,7 @@ impl StepTransactionTrace {
     /// `None`.
     #[allow(dead_code)]
     pub fn step(&mut self) -> Option<&TraceLog> {
-        if self.step > self.inner.struct_logs.len() {
+        if self.step > self.inner.struct_logs.len() - 2 {
```

Comment:
Why `-2`? I'd like to see some description for that.

Reply:
To avoid index out of bounds. The else block does `self.step += 1;` so has to be - 2. Eg. length is 10, index is 8, becomes 9 after incrementing, so it accesses the last element.

Reply:
I mean to add that comment to the code.